sql/core.go

```diff
@@ -707,6 +707,20 @@ type MutableDatabaseProvider interface {
 	DropDatabase(ctx *Context, name string) error
 }
 
+// ExternalStoredProcedureProvider provides access to built-in stored procedures. These procedures are implemented
+// as functions, instead of as SQL statements. The returned stored procedures cannot be modified or deleted.
+type ExternalStoredProcedureProvider interface {
+	// ExternalStoredProcedure returns the external stored procedure details for the procedure with the specified name
+	// that is able to accommodate the specified number of parameters. If no matching external stored procedure is found,
+	// an ErrStoredProcedureDoesNotExist error is returned.
+	ExternalStoredProcedure(ctx *Context, name string, numOfParams int) (*ExternalStoredProcedureDetails, error)
```

Comment:
Consider changing the contract to not require a specific error type for correctness